File: Server/aoRss/Models/Db/ContentModel.vb

Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses

Namespace Models.Db
    Public Class ContentModel        '<------ set set model Name and everywhere that matches this string
        Inherits BaseModel
        'Implements ICloneable
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "Content"      '<------ set content name
        Public Const contentTableName As String = "ccContent"   '<------ set to tablename for the primary content (used for cache names)
        Private Shadows Const contentDataSource As String = "default"             '<------ set to datasource if not default
        '
        '====================================================================================================
        ' -- instance properties

        Public Property AdminOnly As Boolean
        Public Property AllowAdd As Boolean
        Public Property AllowContentChildTool As Boolean
        Public Property AllowContentTracking As Boolean
        Public Property AllowDelete As Boolean
        Public Property AllowMetaContent As Boolean
        Public Property AllowTopicRules As Boolean
        Public Property AllowWorkflowAuthoring As Boolean
        Public Property AuthoringTableID As Integer
        Public Property ContentTableID As Integer
        Public Property DefaultSortMethodID As Integer
        Public Property DeveloperOnly As Boolean
        Public Property DropDownFieldList As String
        Public Property EditorGroupID As Integer
        Public Property IconHeight As Integer
        Public Property IconLink As String
        Public Property IconSprites As Integer
        Public Property IconWidth As Integer
        Public Property InstalledByCollectionID As Integer
        Public Property IsBaseContent As Boolean
        Public Property ParentID As Integer
    End Class
End Namespace

File: Server/aoRss/Models/Db/ContentModel.cs

namespace Contensive.Addons.Rss.Models.Db {
    public class ContentModel : BaseModel {        // <------ set set model Name and everywhere that matches this string
        // Implements ICloneable
        // 
        // ====================================================================================================
        // -- const
        public const string contentName = "Content";      // <------ set content name
        public const string contentTableName = "ccContent";   // <------ set to tablename for the primary content (used for cache names)
        private new const string contentDataSource = "default";             // <------ set to datasource if not default
        public bool AdminOnly { get; set; }
        public bool AllowAdd { get; set; }
        public bool AllowContentChildTool { get; set; }
        public bool AllowContentTracking { get; set; }
        public bool AllowDelete { get; set; }
        public bool AllowMetaContent { get; set; }
        public bool AllowTopicRules { get; set; }
        public bool AllowWorkflowAuthoring { get; set; }
        public int AuthoringTableID { get; set; }
        public int ContentTableID { get; set; }
        public int DefaultSortMethodID { get; set; }
        public bool DeveloperOnly { get; set; }
        public string DropDownFieldList { get; set; }
        public int EditorGroupID { get; set; }
        public int IconHeight { get; set; }
        public string IconLink { get; set; }
        public int IconSprites { get; set; }
        public int IconWidth { get; set; }
        public int InstalledByCollectionID { get; set; }
        public bool IsBaseContent { get; set; }
        public int ParentID { get; set; }
    }
}